spectrochempy/analysis/pls.py

```diff
@@ -0,0 +1,317 @@
+        return self._pls.predict(X, copy)
+
+    def _score(self, X, Y, sample_weight=None):
+        # this method is called by the abstract class score.
+        # Input X, Y, sample_weights are np.ndarray
+        return self._pls.score(X, Y, sample_weight=sample_weight)
+
+    # ----------------------------------------------------------------------------------
+    # Public methods and properties specific to PLS
+    # ----------------------------------------------------------------------------------
+    _docstring.keep_params("analysis_fit.parameters", "X")
+
+    @_docstring.dedent
+    def fit(self, X, Y):
+        """
+        Fit the PLS model on X and Y.
+
+        Parameters
+        ----------
+        %(analysis_fit.parameters.X)s
+
+        Returns
+        -------
+        %(analysis_fit.returns)s
+
+        See Also
+        --------
+        %(analysis_fit.see_also)s
+        """
+        super().fit(X, Y)
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        units=None,
+        title=None,
+        typey="components",
+    )
+    def x_loadings(self):
+        return self._x_loadings
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        meta_from="_Y",
+        units=None,
+        title=None,
+        typey="components",
+    )
+    def y_loadings(self):
+        return self._y_loadings
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        units=None,
+        title=None,
+        typex="components",
+    )
+    def x_scores(self):
+        return self._x_scores
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        meta_from="_Y",
+        units=None,
+        title=None,
+        typex="components",
+    )
+    def y_scores(self):
+        return self._y_scores
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        units=None,
+        title=None,
+        typey="components",
+    )
+    def x_rotations(self):
+        return self._x_rotations
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        meta_from="_Y",
+        typey="components",
+    )
+    def y_rotations(self):
+        return self._y_rotations
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        typey="components",
+    )
+    def x_weights(self):
+        return self._x_weights
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        meta_from="_Y",
+        typey="components",
+    )
+    def y_weights(self):
+        return self._y_weights
+
+    @property
+    def coef(self):
+        coef = NDDataset(self._coef)
+        coef.set_coordset(
+            y=self._Y.x,
+            x=self._X.x,
+        )
+        return coef
+
+    @property
+    @_wrap_ndarray_output_to_nddataset(
+        meta_from="_Y",
+        typesingle="targets",
+    )
+    def intercept(self):
+        return self._intercept
+
+    @property
+    def n_iter(self):
+        return self._n_iter_
+
+    @property
+    def n_features_in(self):
+        return self._n_feature_in
+
```

Comment:
Is it an interesting attribute for spectroChemPy. I know that we wrap the sklearn model, but may be we can simplify (as I think I did for PCA).